simulator/risc_v/riscv_instr.cpp

```diff
@@ -13,103 +13,110 @@
-template<typename I> const auto execute_uret = ALU::jump_and_link<I, ALU::riscv_jr<I>>;
-template<typename I> const auto execute_sret = ALU::jump_and_link<I, ALU::riscv_jr<I>>;
-template<typename I> const auto execute_mret = ALU::jump_and_link<I, ALU::riscv_jr<I>>;
-template<typename I> const auto execute_wfi = do_nothing<I>;
-template<typename I> const auto execute_fence = ALU::addr<I>;
-template<typename I> const auto execute_csrrw = ALU::csrrw<I>;
-template<typename I> const auto execute_csrrs = ALU::csrrs<I>;
-template<typename I> const auto execute_csrrc = do_nothing<I>;
-template<typename I> const auto execute_csrrwi = ALU::csrrwi<I>;
+template<typename I> const auto execute_ecall  = RISCVALU<I>::syscall;
+template<typename I> const auto execute_ebreak = RISCVALU<I>::breakpoint;
+template<typename I> const auto execute_uret   = RISCVALU<I>::template jump_and_link<RISCVALU<I>::riscv_jr>;
+template<typename I> const auto execute_sret   = RISCVALU<I>::template jump_and_link<RISCVALU<I>::riscv_jr>;
+template<typename I> const auto execute_mret   = RISCVALU<I>::template jump_and_link<RISCVALU<I>::riscv_jr>;
+template<typename I> const auto execute_wfi    = do_nothing<I>;
+template<typename I> const auto execute_fence  = RISCVALU<I>::addr;
+template<typename I> const auto execute_csrrw  = RISCVALU<I>::csrrw;
+template<typename I> const auto execute_csrrs  = RISCVALU<I>::csrrs;
+template<typename I> const auto execute_csrrc  = do_nothing<I>;
+template<typename I> const auto execute_csrrwi = RISCVALU<I>::csrrwi;
 template<typename I> const auto execute_csrrsi = do_nothing<I>;
 template<typename I> const auto execute_csrrci = do_nothing<I>;
 // M
-template<typename I> const auto execute_mul = RISCVMultALU::mult_l<I, typename I::RegisterUInt>;
-template<typename I> const auto execute_mulh = RISCVMultALU::mult_h_ss<I, typename I::RegisterUInt>;
-template<typename I> const auto execute_mulhsu = RISCVMultALU::mult_h_su<I, typename I::RegisterUInt>;
-template<typename I> const auto execute_mulhu = RISCVMultALU::mult_h_uu<I, typename I::RegisterUInt>;
-template<typename I> const auto execute_div = RISCVMultALU::div<I, sign_t<typename I::RegisterUInt>>;
-template<typename I> const auto execute_divu = RISCVMultALU::div<I, typename I::RegisterUInt>;
-template<typename I> const auto execute_rem = RISCVMultALU::rem<I, sign_t<typename I::RegisterUInt>>;
-template<typename I> const auto execute_remu = RISCVMultALU::rem<I, typename I::RegisterUInt>;
+template<typename I> const auto execute_mul    = RISCVMultALU<I>::mult_l;
+template<typename I> const auto execute_mulh   = RISCVMultALU<I>::mult_h_ss;
+template<typename I> const auto execute_mulhsu = RISCVMultALU<I>::mult_h_su;
+template<typename I> const auto execute_mulhu  = RISCVMultALU<I>::mult_h_uu;
+template<typename I> const auto execute_div    = RISCVMultALU<I>::template div<sign_t<typename I::RegisterUInt>>;
+template<typename I> const auto execute_divu   = RISCVMultALU<I>::div;
```

Comment:
Explicit type specification is better, as it makes the difference between instructions obvious. Please put back.